### ISSUE TITLE
Finalizing website and paper

### ISSUE THREAD

**Issue by Buyun-Liang** [sun-umn/PyGRANSO#30]
I'll try to wrap up the website and the paper today. Please visit https://github.com/sun-umn/NCVX_Documentation for the source code of the website

**Comment by t-mitchell** [comment]
Just ping me when are done editing and want me to review either the website or paper.

**Comment by t-mitchell** [comment]
BTW, for the GRANSO code example in the paper, I suggest making that also use the `combined_fn` format, since (a) it'll be a little shorter (and we need space) and (b) that's the recommended GRANSO interface anyway.  ;-)

**Comment by Buyun-Liang** [comment]
PyGRANSO website ready for review: https://ncvx.org/PyGRANSO/

**Comment by Buyun-Liang** [comment]
Paper updated. Ju will review the paper with me tomorrow.

**Comment by t-mitchell** [comment]
> Paper updated. Ju will review the paper with me tomorrow.

Great, and I just took a look.  The main issue I see is that we will need to find things to cut or make more compact to get back to 4 pages before references.  If you and Ju want to discuss over Zoom, let me know.  I am pretty flexible.

Once the paper is submitted and updated arXiv, I guess you should update any other places you posted about the code/preprint, e.g., https://pythonrepo.com/repo/sun-umn-ncvx-python-deep-learning

**Comment by Buyun-Liang** [comment]
Maybe I will meet with Ju first, since I don't know when he has time today.

We only post the paper on arxiv. All the other websites should be made by machine.

**Comment by t-mitchell** [comment]
Always remember to set the directory variables in the tutorials to your specific PyGRANSO installation location and the folder containing the data for the examples.  (Too wordy but you can edit to make it better)

**Comment by Buyun-Liang** [comment]
Just updated the JMLR paper with Ju. We can have a zoom meeting tomorrow for the final round review if you have time.

**Comment by t-mitchell** [comment]
I'll look at it now and see if I find anything.

**Comment by t-mitchell** [comment]
Just read it.  Overall it is great, but I did notice a handful of grammar issues / typos and one issue about the two QP solves per iteration (which I had deleted earlier).  Shall I just make edits to the paper now with my \tim macro?   

Regarding the two QP solves, I had changed that because it's not quite correct.  GRANSO can do more than 2 QP solves per iteration, e.g., when it lowers the value of the penalty parameter mu.  Also, if there are no constraints, it generally solves 1 QP per iteration.  I could say "two crucial TYPES OF quadratic problems" but that may cause a new line to be added, which then might mean we don't fit on 4 pages.

There are also some minor problems in the bib entries.  I'll open another issue for these and we can decide what to do on those.

**Comment by Buyun-Liang** [comment]
Yes, please edit it with your macro. Do you have time for zoom discussion tomorrow?

**Comment by t-mitchell** [comment]
Some minor bibtex stuff:

In the paper:
- [ ] A lot of title fields don't have proper capitalization, i.e., many things are incorrectly in lower case: riemannian, pytorch, Cvx, etc.  I already added {} to BFGS-SQP in the bibtex in my paper.  Not sure what you and Ju prefer on the rest.  Indeed, it's annoying busy work.

In the readme files, website, code, etc, with actual bibtex for users to copy/paste, we should fix:
- [ ] liang2021ncvx:  `{{NCVX}: {A} user-friendly ...}` (I'm not sure anything else needs capitalization but up to you)
- [ ] curtis2017bfgs: `{A {BFGS-SQP} ...}` 
- [ ] lastly, can we change curtis2017bfgs to curtis2017bfgssqp?  The actual BFGS people might not like curtis2017bfgs. ;-)
But this should only be a handful of files to update ase we don't use bibtex formatting in any of the .py files.

**Comment by t-mitchell** [comment]
> Yes, please edit it with your macro. Do you have time for zoom discussion tomorrow?

Yes, I am fairly available tomorrow.  I will make edits right now.

**Comment by Buyun-Liang** [comment]
Meeting invitation sent

**Comment by t-mitchell** [comment]
I am done editing the paper.  I was able to correct the QP thing while staying at 4 pages.  All of my changes are marked with the `\tim` macro except:
- I deleted a handful of commas that were incorrect
- I enlarged the screen shots from width=0.90 to 0.96

I did not do any edits to the bibtex besides the {BFGS-SQP} change.

**Comment by t-mitchell** [comment]
> Some minor bibtex stuff:
> 
> In the paper:
> 
> * [ ]  A lot of title fields don't have proper capitalization, i.e., many things are incorrectly in lower case: riemannian, pytorch, Cvx, etc.  I already added {} to BFGS-SQP in the bibtex in my paper.  Not sure what you and Ju prefer on the rest.  Indeed, it's annoying busy work.
> 
> In the readme files, website, code, etc, with actual bibtex for users to copy/paste, we should fix:
> 
> * [ ]  liang2021ncvx:  `{{NCVX}: {A} user-friendly ...}` (I'm not sure anything else needs capitalization but up to you)
> * [ ]  curtis2017bfgs: `{A {BFGS-SQP} ...}`
> * [ ]  lastly, can we change curtis2017bfgs to curtis2017bfgssqp?  The actual BFGS people might not like curtis2017bfgs. ;-)
>   But this should only be a handful of files to update ase we don't use bibtex formatting in any of the .py files.

BTW, I am happy to do the work of editing all this bibtex stuff in the paper and elsewhere.  But I didn't want to start changing things without your okay to do so.

**Comment by Buyun-Liang** [comment]
I made several updates:
1. ReadMe.md in Source Code: update introduction and citation. 
2. Website: update description in https://ncvx.org home page and also updated introduction section and citation of PyGRANSO page https://ncvx.org/PyGRANSO

**Comment by Buyun-Liang** [comment]
There is something wrong with the dependency installation file. I'll try to fix it today, but the JMLR submission may be delayed.

We can update the arxiv first, which does not require source code

**Comment by Buyun-Liang** [comment]
Just fixed the dependency issue and made a new initial public release.
